### PR TITLE
WIP: working docker setup

### PR DESCRIPTION
**Pull request by jguice**
This *probably* only works on Apple Silicon (ARM) macOS.

### CODE REVIEW

File: first-bank-of-change/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      dockerfile: npm.Dockerfile
+    volumes:
+      - .:/var/opt/app
+    working_dir: /var/opt/app
+    ports:
+      - "4200:4200"
+  selenium:
+    networks:
+      - bdd
+    image: seleniarm/standalone-chromium:114.0
```
Comment (jguice) [inline first-bank-of-change/docker-compose.yaml:20]: Need to test the [x86 image](https://hub.docker.com/layers/selenium/standalone-chrome/114.0/images/sha256-1b809a961a0a77787a7cccac74ddc5570b7e89747f925b8469ddb9a6624d4ece?context=explore) on a non-Apple Silicon system

**Comment by drewkhoury** [comment]
**How to test:** Clone this repo and checkout the branch:

```git clone https://github.com/jguice/cucumber-declarative-gherkin.git && git checkout feat/3musketeers```

---

**_Instructions to add to README..._**

From the `cucumber-declarative-gherkin` folder, ie after you've done a clone and `cd cucumber-declarative-gherkin`:

- Run the containerized selenium driver: `docker-compose up selenium`

- Open your browser: http://localhost:7900/?autoconnect=1&resize=scale&password=secret

- Run the end to end tests: `make npm run e2e`

See https://www.youtube.com/watch?v=3sf8--xIU0U for demo.

**Comment by drewkhoury** [comment]
@jguice works for me on apple silicon

File: first-bank-of-change/docker-compose.yaml (hunk 1 above)
Comment (jguice) [reply first-bank-of-change/docker-compose.yaml:20]: Actually, the one above works on x86 too!

**Comment by jguice** [comment]
#### Updated instructions
1. After clone, `cd first-bank-of-change`
2. Run `make npm install`
3. Run `docker compose up selenium`
4. Open your browser to http://localhost:7900/?autoconnect=1&resize=scale&password=secret
5. In another shell session (in `first-bank-of-change`), `make npm run e2e`

You should see the tests run against the browser in the VNC viewer.  The run should fail due to one failing test, but otherwise all tests should execute.

**Comment by drewkhoury** [comment]
> #### Updated instructions
> 1. After clone, `cd first-bank-of-change`
> 2. Run `make npm install`
> 3. Run `docker compose up selenium`
> 4. Open your browser to http://localhost:7900/?autoconnect=1&resize=scale&password=secret
> 5. In another shell session (in `first-bank-of-change`), `make npm run e2e`
> 
> You should see the tests run against the browser in the VNC viewer. The run should fail due to one failing test, but otherwise all tests should execute.

Can you update the readme to flow so people know about these changes.

I think `./quick-notes.md` is not required at all now as your containerized steps replace everything here (can remove).

I would update `/README.md` to have your notes under **START HERE** section.

I would say that some sections could be removed now, like **Where to Start?** and **Perquisites** which are replaced by 3m.

**Pull request merged**